simple_parsing/helpers/partial.py

```diff
@@ -0,0 +1,308 @@
+            field_type = class_annotations[name]
+        elif default is not dataclasses.MISSING:
+            # Infer the type from the default value.
+            # try:
+            # # BUG: There is a default of '<required parameter>'.
+            # if str(default) == "<required parameter>":
+            #     breakpoint()
+            field_type = infer_type_annotation_from_default(default)
+            # except:
+            #     field_type = Any
+        else:
+            logger.warning(
+                f"Don't know what the type of field '{name}' of class {cls} is! "
+                f"Ignoring this argument."
+            )
+            continue
+
+        class_help_entries = {v for k, v in class_docstring_help.items() if k.startswith(name)}
+        init_help_entries = {v for k, v in class_constructor_help.items() if k.startswith(name)}
+        help_entries = init_help_entries or class_help_entries
+        if help_entries:
+            help_str = help_entries.pop()
+        else:
+            help_str = ""
+
+        if default is dataclasses.MISSING:
+            field = simple_parsing.field(help=help_str, required=True)
+            # insert since fields without defaults need to go first.
+            fields.insert(0, (name, field_type, field))
+            logger.debug(f"Adding required field: {fields[0]}")
+        else:
+            field = simple_parsing.field(default=default, help=help_str)
+            fields.append((name, field_type, field))
+            logger.debug(f"Adding optional field: {fields[-1]}")
+
+    cls_name = _get_generated_config_class_name(cls)
+    config_class = make_dataclass(cls_name=cls_name, bases=(Partial,), fields=fields)
+    config_class._target_ = cls
+    config_class.__doc__ = (
+        f"Auto-Generated configuration dataclass for {cls.__module__}.{cls.__qualname__}\n"
+        + (cls.__doc__ or "")
+    )
+
+    return config_class
+
+
+def _parse_args_from_docstring(docstring: str) -> dict[str, str]:
+    """Taken from `pytorch_lightning.utilities.argparse`."""
+    arg_block_indent = None
+    current_arg = ""
+    parsed = {}
+    for line in docstring.split("\n"):
+        stripped = line.lstrip()
+        if not stripped:
+            continue
+        line_indent = len(line) - len(stripped)
+        if stripped.startswith(("Args:", "Arguments:", "Parameters:")):
+            arg_block_indent = line_indent + 4
+        elif arg_block_indent is None:
+            continue
+        elif line_indent < arg_block_indent:
+            break
+        elif line_indent == arg_block_indent:
+            current_arg, arg_description = stripped.split(":", maxsplit=1)
+            parsed[current_arg] = arg_description.lstrip()
+        elif line_indent > arg_block_indent:
+            parsed[current_arg] += f" {stripped}"
+    return parsed
+
+
+try:
+    # This only seems to be necessary for the SGD optimizer.
+    from torch.optim.optimizer import _RequiredParameter
```

Comment:
What's the story here? There's a single exception only for `torch.optim`?

Reply:
Added a better explanation in here:
https://github.com/lebrice/SimpleParsing/blob/7844d4ed916f2095cecdc921f77213e35da20d2b/simple_parsing/helpers/partial.py#L37-L56

This basically applies a fix that makes it possible to create the Partial classes for the torch optimizer types. 

There's one issue with this though: If pytorch is installed in the environment, it gets loaded here, which can be quite slow.  I'm not 100% sure of the best way to handle this atm.

Do you have any suggestions?